invoicer_project/static/js/add_client_request.js

```diff
@@ -0,0 +1,286 @@
+    }
+    try {
+        const response = await fetch(url, {
+            headers: headers,
+            body: data,
+            method: "POST"
+        });
+        status = response.status;
+        console.log(`Status code: ${status}`);
+
+    } catch (error) {
+        console.error(error);
+    }
+    return status;
+}
+
+async function actionBasedOnStatusCode(statusCode, data) {
+    if (statusCode === 201) {
+        for (let field of returnAllFields()) {
+            field.value = '';
+        }
+        window.location.href = host + "/clients/home/";
+    } else if (statusCode === 401) {
+        const obtainedNewTokens = await obtainNewAccessToken();
+        if (!obtainedNewTokens) {
+            window.location.href = host + '/user/login/';
+        } else {
+            const status = await sendAddUserRequest(host + "/clients/client/", data);
+            actionBasedOnStatusCode(status, data);
+        }
+        console.log("Updating token");
+    } else if (statusCode === 400) {
+        console.log("Backend validation error: 400 status");
+    } else {
+        console.log(`Unknown error: status code = ${statusCode}`);
+    }
+}
+
+async function obtainNewAccessToken() {
+    let response;
+    const data = {refresh: window.localStorage.getItem('refreshToken')};
+    try {
+        response = await fetch(host + '/user/refresh/', {
+            method: "POST",
+            body: JSON.stringify(data)
+        });
+        const newToken = await response.json();
+        const accessToken = newToken['access'];
+        window.localStorage.setItem('accessToken', accessToken);
+    } catch (error) {
+        console.error(error);
+    }
+    return response.status === 200;
+}
+
+document.getElementById("request_sender").addEventListener("click", async () => {
+    const validationFieldsList = validateClientAdd();
+    if (allAreFalse(validationFieldsList)) {
+        const data = JSON.stringify({
+            first_name: nameField.value,
+            last_name: surnameField.value,
+            email: emailField.value,
+            phone_number: telephoneField.value,
+            zip_code: zipField.value,
+            country: countryField.value,
+            city: cityField.value,
+            address: addressField.value
+        });
+        const serverResponseStatus = await sendAddUserRequest(host + "/clients/client/", data);
+        actionBasedOnStatusCode(serverResponseStatus, data);
+    } else {
+        setErrorAttributesToFields(validationFieldsList);
+    }
+});
+
+function fillInitials(userData) {
+    const userFirstName = userData["first_name"];
+    const userLastName = userData["last_name"];
+    document.getElementById("user_name").textContent = userFirstName + " " + userLastName;
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+    obtainUserInitials();
+});
+async function obtainUserInitials() {
+    let responseCode;
+    if (window.localStorage.getItem('accessToken'))
```

Comment:
```suggestion
    if (window.localStorage.getItem('accessToken')){
```

Reply:
didn't get the idea, but i think you wanted to create constant for token to not to call .getItem twice from localstorage like this ?
![image](https://user-images.githubusercontent.com/99320958/221381476-4a97aed9-e54d-4fdf-8e8e-34aeb587b164.png)